tests/core/test_version.py

```diff
@@ -16,10 +17,16 @@ def test_version_working():
 
 # When adding more of these, ensure they are added in ascending order of precedence
 version_tests = (
+    "3.0.0.dev1",
+    "3.0.0.dev2",
+    "3.0.0a32.dev12",
+    "3.0.0a32",
     "3.0.0a32.post10.dev12",
+    "3.0.0a32.post10",
     "3.0.0rc1.dev1",
```

Comment:
Add one `b`eta string to the tests, since that is technically supported but also not currently tested, and this is good to go.